spec/models/client_spec.rb

```diff
@@ -23,6 +23,46 @@
     end
   end
 
+  describe "Client transfer" do
+    let!(:prefix)  { create(:prefix) }
+    let!(:prefixes)  { create(:client_prefix, client: client, prefix: prefix) }
+    let!(:prefixes_dos)  { create(:provider_prefix, provider: provider, prefix: prefix) }
+    let(:new_provider) { create(:provider, symbol: "QUECHUA") }
+    let(:options) { { target_id: new_provider.symbol } }
+    let(:bad_options) { { target_id: "SALS" } }
+
+    it "works" do
+      client.transfer(options)
+
+      expect(client.provider_id).to eq(new_provider.symbol.downcase)
+      expect(new_provider.prefix_ids).to include(prefix.uid)
+      expect(provider.prefix_ids).not_to include(prefix.uid)
```

Comment:
Add a check for all other prefixes of the old provider. I worry that the above code deletes them.

Reply:
a test has been included now